src/PriorityQueue.php

```diff
@@ -292,19 +305,20 @@ public function toArray($flag = self::EXTR_DATA)
      * Please see {@link getIterator()} for details on the necessity of an
      * internal queue class. The class provided should extend SplPriorityQueue.
      *
-     * @param  string $class
-     * @return PriorityQueue
+     * @param  class-string<\SplPriorityQueue> $class
+     * @return $this
      */
     public function setInternalQueueClass($class)
     {
+        /** @psalm-suppress RedundantCastGivenDocblockType */
```

Comment:
If we remove the cast elsewhere, we can remove it also here: perhaps best to keep casts across the patch, to avoid regressions